src/Core/Signers/SignerV4.php

```diff
@@ -214,31 +159,38 @@ private function prepareBody(Request $request, string $amzDate, string $credenti
         $fullChunkCount = $chunkCount * self::CHUNK_SIZE === $contentLength ? $chunkCount : ($chunkCount - 1);
 
         $metaLength = \strlen(";chunk-signature=\r\n\r\n") + 64;
-        $contentLength = $contentLength + $fullChunkCount * ($metaLength + \strlen((string) dechex(self::CHUNK_SIZE))) + ($chunkCount - $fullChunkCount) * ($metaLength + \strlen((string) dechex($contentLength % self::CHUNK_SIZE))) + $metaLength + 1;
-        $request->setHeader('content-length', $contentLength);
-
-        $last = false;
-        $streamBody = static function () use ($streamReader, $amzDate, $credentialScope, &$signature, &$last, $signingKey): string {
-            if ($last) {
-                return '';
+        $request->setHeader('content-length', $contentLength + $fullChunkCount * ($metaLength + \strlen((string) dechex(self::CHUNK_SIZE))) + ($chunkCount - $fullChunkCount) * ($metaLength + \strlen((string) dechex($contentLength % self::CHUNK_SIZE))) + $metaLength + 1);
+
+        if (!$body instanceof FixedSizeStream) {
+            $body = new FixedSizeStream($body, self::CHUNK_SIZE);
+        }
```

Comment:
These 3 lines could be removed if we were using a named factory. Either: 

```php
$body = StreamFactory::fixedSizeStream($body);
```
or
```php
$body = FixedSizeStream::create($body);
```

Reply:
good idea